Userland/Services/LaunchServer/Launcher.cpp

```diff
@@ -233,6 +241,16 @@ bool Launcher::open_with_user_preferences(HashMap<String, String> const& user_pr
     if (program_path.has_value())
         return spawn(program_path.value(), arguments);
 
+    String executable = "";
+    if (for_each_handler(key, user_preferences, [&](auto const& handler) -> bool {
+            if (executable.is_empty() && (handler.mime_types.contains(key) || handler.file_types.contains(key) || handler.protocols.contains(key))) {
+                executable = handler.executable;
+                return true;
+            }
+            return false;
+        }))
+        return spawn(executable, arguments);
```

Comment:
Nit-pick: Wrapping this in `{}` would make it easier to understand. It's quite far from the `if`.